ast/compile.go

```diff
@@ -3937,6 +3945,30 @@ func (s localDeclaredVars) GlobalOccurrence(x Var) (varOccurrence, bool) {
 	return newVar, false
 }
 
+// Seen marks x as seen by incrementing its counter
+func (s localDeclaredVars) Seen(x Var) {
```

Comment:
I guess this can't be called concurrently but just wondering.

Reply:
In this case, I don't think that would improve performance enough to warrant the added complexity.

Reply:
The compiler runs everything sequentially, so it's not concurrency-safe in many places, I'm afraid.